lib/active_model_serializers/adapter/json_api/pagination_links.rb

```diff
@@ -20,12 +20,13 @@ def initialize(collection, adapter_options)
         end
 
         def as_json
-          per_page = collection.try(:per_page) || collection.try(:limit_value) || collection.size
-          pages_from.each_with_object({}) do |(key, value), hash|
-            params = query_parameters.merge(page: { number: value, size: per_page }).to_query
-
-            hash[key] = "#{url(adapter_options)}?#{params}"
-          end
+          {
+            "self":  location_url,
+            "first": first_page_url,
+            "prev":  prev_page_url,
+            "next":  next_page_url,
+            "last":  last_page_url
+          }
```

Comment:
I think this looks much nicer :)  probably should be `self:` vs. `"self":`.. not sure why I used quotes... not asking for a change.. just thinking aloud